### PR TITLE
chore: update CHANGELOG for v2.16.0

### PR DESCRIPTION
**Pull request by dbolduc**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12783)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -123,17 +123,38 @@ the APIs in these libraries are stable, and are ready for production use.
...
+### [OAuth2](/google/cloud/oauth2/README.md)
+
+- feat(oauth2): add quickstart, README, etc. ([#12754](https://github.com/googleapis/google-cloud-cpp/pull/12754))
+
 ### [OpenTelemetry](/google/cloud/opentelemetry/README.md)
 
-Features to instrument the Google Cloud C++ libraries for [distributed tracing]
-using [OpenTelemetry] are now GA. See the [quickstart][otel-quickstart] within
-the library to learn more about these tools.
+The Google Cloud C++ libraries have been instrumented for [distributed tracing]
```
Comment (coryan) [inline CHANGELOG.md:132]: The passive voice shall never be used...

Consider:

```
We instrumented the Google Cloud C++ libraries for [distributed tracing]
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -123,17 +123,38 @@ the APIs in these libraries are stable, and are ready for production use.
...
 
 ### [Policy Troubleshooter](/google/cloud/policytroubleshooter/README.md)
 
 Remove `PoliciesClient`. This client was placed in the wrong library. The
 correct location is google/cloud/iam/v2.
 
+### [Storage](/google/cloud/storage/README.md)
+
+- fix(GCS+gRPC): race condition in async `Insert()` ([#12715](https://github.com/googleapis/google-cloud-cpp/pull/12715))
+- docs(GCS+gRPC): additional `ReadObject()` example ([#12695](https://github.com/googleapis/google-cloud-cpp/pull/12695))
```
Comment (coryan) [inline CHANGELOG.md:147]:
```suggestion
```

**Review by scotthart — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12783)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dbolduc)

**Review by coryan — CHANGES_REQUESTED** [review]

**Review by devbww — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (dbolduc) [reply CHANGELOG.md]: Done.

**Review by coryan — APPROVED** [review]

**Review by coryan — APPROVED** [review]

**Pull request merged**